internal/telemetry/tracker.go

```diff
@@ -26,15 +30,24 @@ func (tracker *stdoutTracker) Track(event event) {
 	)
 }
 
-type segmentTracker struct{}
+type segmentTracker struct {
+	client analytics.Client
+	logger *log.Logger
+}
+
+func newSegmentTracker(writeKey string, logger *log.Logger) *segmentTracker {
```

Comment:
ok, so here is where you're going to want:

```go
var (
  segmentWriteKey = "" // will substitute value at build time with proper key
)
```

and then you can just refer to that when you call `analytics.New`

Reply:
still trying to think about what we want to do to protect ourselves if this build substitution doesn't happen properly ... 

i think we maybe re-introduce that `error` return value in the telemetry service constructor, but lets discuss this more offline since you have a better idea of the nuances here.  i just want to make sure failed telemetry doesn't prevent the CLI from running at all

Reply:
I didn't reintroduce the error for the constructor for now and made it return another tracker instead, since it's not really an error that we should fail the CLI for. I'm unsure how the `stdout` tracker would be used (other than maybe dev), but maybe we could return that instead of the `segment` tracker

Reply:
the `stdout` tracker may go away all together... its noisy and and not all that helpful for anyone involved haha (it was cool during early developemnt).

i like the idea of just transitioning over to use the `noopTracker` if we find that `segmentWriteKey` is empty, let's roll with that